Adding thalysantana-appcenter-datasource v1.0.0

thalysantana opened the pull request:
A User API Token is required to connect to the App Center. 
Install instructions provided on README.md. 
Plugin signed and packaged as required by Grafana's team.

marcusolsson commented:
Looks great! 🎉 I'll add it to the review queue!

marcusolsson commented:
I've had the chance to review the plugin. I didn't know about App Center, but it looks really cool!

I first tried the plugin on 7.3.6, with limited success. It seems like the plugin doesn't yet support 7.3.x. 
<img width="546" alt="Screen Shot 2021-01-29 at 15 07 59" src="https://user-images.githubusercontent.com/8396880/106285348-a7f3c100-6244-11eb-9cf3-def96397c331.png">

I can see in your plugin.json that you only support 7.2.x (rather than >=7.2.0 or 7.x.x), so I if this is intentional I won't ask you to change at this point. In the README, it seems like you intend to support later versions as well. Please confirm what versions of Grafana you're targeting with this plugin.

I was able to get all query types except "apps" to work. When I selected "apps", the query seems to run indefinitely. Since the other queries works, I don't think it's related to my internet connection. Is there any other reason why this query doesn't complete?

Minor suggestions that I'll leave up to you to decide if you want to change:

- I would consider using a more user-friendly label for the query type, i.e. "Error groups" rather than "errorGroups"
- Grafana uses sentence case for headings and other labels. If you change "Query Type" to "Query type" it will be more consistent with the rest of Grafana

Finally, I'd recommend that you add to your documentation whether I need a Full Access token or Read Only token (I assumed I could use the Read Only one).

thalysantana commented:
Hi  Marcus.


   - Labels - I've updated them to follow the pattern you suggested.
   - Token access type - I've changed the readme file to mention that a
   Full Access token is required
   - Apps query - Please test it using a full access token. App Center
   requires it to run this query.
   - Image Error - I wasn't able to reproduce the error. Could you give
   more information about it?
   - Grafana 7.3.x issues - I've been testing using Grafana v7.3.7
   (1e261642f4) and haven't found any problems. Could you provide more
   information about the errors you got?


   - Grafana v7.3.7 (1e261642f4)

About the supported version, the Grafana Toolkit added the 7.2.x to the
plugin.json when I created the project, but I've been using it on 7.3.7
without any issues. How can I be sure it works on both versions?

I will wait for your answer so I can fix all the issues and push
the changes to Github.
Thanks for your time and patience to review this.

King regards,

Thálys Santana
Skype: thalysantana


On Sat, Jan 30, 2021 at 3:27 AM Marcus Olsson <notifications@github.com>
wrote:

> I've had the chance to review the plugin. I didn't know about App Center,
> but it looks really cool!
>
> I first tried the plugin on 7.3.6, with limited success. It seems like the
> plugin doesn't yet support 7.3.x.
> [image: Screen Shot 2021-01-29 at 15 07 59]
> <https://user-images.githubusercontent.com/8396880/106285348-a7f3c100-6244-11eb-9cf3-def96397c331.png>
>
> I can see in your plugin.json that you only support 7.2.x (rather than
> >=7.2.0 or 7.x.x), so I if this is intentional I won't ask you to change at
> this point. In the README, it seems like you intend to support later
> versions as well. Please confirm what versions of Grafana you're targeting
> with this plugin.
>
> I was able to get all query types except "apps" to work. When I selected
> "apps", the query seems to run indefinitely. Since the other queries works,
> I don't think it's related to my internet connection. Is there any other
> reason why this query doesn't complete?
>
> Minor suggestions that I'll leave up to you to decide if you want to
> change:
>
>    - I would consider using a more user-friendly label for the query
>    type, i.e. "Error groups" rather than "errorGroups"
>    - Grafana uses sentence case for headings and other labels. If you
>    change "Query Type" to "Query type" it will be more consistent with the
>    rest of Grafana
>
> Finally, I'd recommend that you add to your documentation whether I need a
> Full Access token or Read Only token (I assumed I could use the Read Only
> one).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/grafana/grafana-plugin-repository/pull/799#issuecomment-769836419>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABEZFARMFHDU6IVL7XOSH7TS4LAULANCNFSM4UDGD2BA>
> .
>

marcusolsson commented:
I tested the plugin again, and it seems like the `meta` of null is due to the lack of a default query type. Once I select a query, it works.

Also, I now noticed that the plugin logs the instance settings. Could you remove this line?

https://github.com/thalysantana/grafana-appcenter-datasource/blob/875041934274c51effee10f0ad8609e32f8dd0d4/src/DataSource.ts#L26

The `grafanaDependency` property will be matched against the currently running Grafana version to determine whether the author supports that version or not. It doesn't guarantee that it will work. If you don't yet know what versions you support, I suggest that started with something like `>=7.0.0` which will let users install your plugin as long as they at least run 7.0.0.

thalysantana commented:
Hi Marcus.
I have made the modifications you suggested and built the release. I am
just not sure about how I should proceed.
Should I create a new merge request to grafana-plugin-repository, or just
fix the code on the current endpoint (
https://github.com/thalysantana/grafana-appcenter-datasource/releases/download/v1.0.0/thalysantana-appcenter-datasource-1.0.0.zip)?


Kind regards,

Thálys Santana
Skype: thalysantana


On Tue, Feb 9, 2021 at 2:53 AM Marcus Olsson <notifications@github.com>
wrote:

> I tested the plugin again, and it seems like the meta of null is due to
> the lack of a default query type. Once I select a query, it works.
>
> Also, I now noticed that the plugin logs the instance settings. Could you
> remove this line?
>
>
> https://github.com/thalysantana/grafana-appcenter-datasource/blob/875041934274c51effee10f0ad8609e32f8dd0d4/src/DataSource.ts#L26
>
> The grafanaDependency property will be matched against the currently
> running Grafana version to determine whether the author supports that
> version or not. It doesn't guarantee that it will work. If you don't yet
> know what versions you support, I suggest that started with something like
> >=7.0.0 which will let users install your plugin as long as they at least
> run 7.0.0.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/grafana/grafana-plugin-repository/pull/799#issuecomment-775165711>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABEZFASEMYANT2TN2OPQOFDS57UFVANCNFSM4UDGD2BA>
> .
>

marcusolsson commented:
You can keep this PR, and the URL if you'd like, but please update the commit and the hash for the new release 🙏

marcusolsson commented:
Changes looks good, but now there seems to be a missing screenshot. If it has been removed, it should be removed from the plugin.json as well.

<img width="579" alt="CleanShot 2021-02-11 at 15 56 19@2x" src="https://user-images.githubusercontent.com/8396880/107653256-b6e56500-6c81-11eb-8818-eda6df123c7a.png">

thalysantana commented:
Hi Marcus,
I've fixed the issues with the screenshots.

Kind regards

Thálys Santana
Skype: thalysantana


On Fri, Feb 12, 2021 at 3:58 AM Marcus Olsson <notifications@github.com>
wrote:

> Changes looks good, but now there seems to be a missing screenshot. If it
> has been removed, it should be removed from the plugin.json as well.
>
> [image: CleanShot 2021-02-11 at 15 56 19@2x]
> <https://user-images.githubusercontent.com/8396880/107653256-b6e56500-6c81-11eb-8818-eda6df123c7a.png>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/grafana/grafana-plugin-repository/pull/799#issuecomment-777536109>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABEZFAX3Q3CZ5BFLJVDHYULS6PWCTANCNFSM4UDGD2BA>
> .
>

marcusolsson commented:
Published! 🎉

thalysantana commented:
Thanks a lot. 🙌

On Fri, 12 Feb 2021, 21:25 Marcus Olsson, <notifications@github.com> wrote:

> Published! 🎉
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/grafana/grafana-plugin-repository/pull/799#issuecomment-778051816>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABEZFAQNQQONTI7PQGOWAELS6TQXBANCNFSM4UDGD2BA>
> .
>
